### PR TITLE
bug 1440764: add refresh button in the worker types explorer

### PR DESCRIPTION
**Pull request by YinYue33**
I added a button to rerender the whole page on  https://tools.taskcluster.net/provisioners/aws-provisioner-v1/worker-types

### CODE REVIEW

**Comment by YinYue33** [comment]
Sure, I will work on this today

**Comment by YinYue33** [comment]
I added a props to WorkerTypeTable to keep track of clicking. Now workerTypeTable will update either when provisionerID changes or when refresh button is clicked.

File: src/views/WorkerTypes/WorkerManager.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ export default class WorkerManager extends React.PureComponent {
       orderBy: null,
       actions: [],
       error: null,
+      clickState: false,
```
Comment (helfi92) [inline src/views/WorkerTypes/WorkerManager.js]: Can we rename this to something like `refreshFlag`.

File: src/views/WorkerTypes/WorkerManager.js [hunk 2]
```diff
@@ -218,6 +223,9 @@ export default class WorkerManager extends React.PureComponent {
                 </ToggleButton>
               </ToggleButtonGroup>
             </ButtonToolbar>
+            <Button onClick={this.handleRefresh} bsSize="sm">
```
Comment (helfi92) [inline src/views/WorkerTypes/WorkerManager.js]: Having the button outside `<ButtonToolbar>` will have the refresh button displayed on a new line. I would like to see the refresh button be aligned with the rest of the buttons.

File: src/views/WorkerTypes/WorkerManager.js (hunk 1 above)
Comment (eliperelman) [reply src/views/WorkerTypes/WorkerManager.js]: I think using a boolean here also gives a false impression about needing to refresh or not. Let's use a numeric increment instead:

```js
handleRefresh = () => {
  this.setState({ refreshId: ++this.state.refreshId });
};
```

Comment (YinYue33) [reply src/views/WorkerTypes/WorkerManager.js]: Good idea. But will this potentially exceed largest integer as the server runs?

**Pull request merged**